ui/src/components/Main.jsx

```diff
@@ -19,144 +16,70 @@ import WorkflowParametersDialog from '../containers/WorkflowParametersDialog';
 import GphlWorkflowParametersDialog from '../containers/GphlWorkflowParametersDialog';
 import SelectProposalContainer from '../containers/SelectProposalContainer';
 import diagonalNoise from '../img/diagonal-noise.png';
-import { resetChatMessageCount } from '../actions/remoteAccess';
-import { Widget, addResponseMessage, addUserMessage } from 'react-chat-widget';
 import { showDialog } from '../actions/general';
 import { LimsResultDialog } from './Lims/LimsResultDialog';
 import LoadingScreen from './LoadingScreen/LoadingScreen';
 import styles from './Main.module.css';
 
 import 'react-chat-widget/lib/styles.css';
 import './rachat.css';
-import { store } from '../store';
 import { getInitialState } from '../actions/login';
-import { fetchChatMessages, sendChatMessage } from '../api/remoteAccess';
 import MXNavbar from './MXNavbar/MXNavbar';
+import ChatWidget from './ChatWidget';
 
-class Main extends React.Component {
-  constructor(props) {
-    super(props);
-    this.handleClick = this.handleClick.bind(this);
-    this.handleNewUserMessage = this.handleNewUserMessage.bind(this);
-    this.onChatContainerClick = this.onChatContainerClick.bind(this);
-  }
-
-  componentDidMount() {
-    if (!this.props.applicationFetched) {
-      store.dispatch(getInitialState());
-    }
+function Main() {
+  const dispatch = useDispatch();
+  const { pathname } = useLocation();
 
-    // eslint-disable-next-line promise/prefer-await-to-then, promise/catch-or-return
-    fetchChatMessages().then((json) => {
-      json.messages.forEach((entry) => {
-        if (entry.username === this.props.login.user.username) {
-          addUserMessage(`${entry.date} **You:** \n\n ${entry.message} \n\n`);
-        } else {
-          addResponseMessage(
-            `${entry.date} **${entry.nickname}:** \n\n ${entry.message}`,
-          );
-        }
-      });
-    });
-  }
+  const inControl = useSelector((state) => state.login.user.inControl);
+  const general = useSelector((state) => state.general);
 
-  onChatContainerClick() {
-    this.props.resetChatMessageCount();
-  }
+  useEffect(() => {
+    dispatch(getInitialState());
+  }, [dispatch]);
 
-  handleNewUserMessage(message) {
-    sendChatMessage(message, this.props.login.user.username);
-  }
+  const showReadOnlyDiv =
+    !inControl && pathname !== '/remoteaccess' && pathname !== '/help';
 
-  handleClick(e) {
-    e.preventDefault();
-    e.stopPropagation();
-    e.nativeEvent.stopImmediatePropagation();
```

Comment:
I've removed this handler from the read-only `div`. I don't think it's needed, since the read-only `div` is on top of the main part of the page and it's a stand-alone element with no children, so it already prevents pointer events from reaching any of the elements it covers. It would be needed if the backdrop was implemented as a CSS pseudo-element on `div.main`, for instance, since pointer events would then bubble down to the `div`'s children.